### PR TITLE
Add more information to Retry/Timeout logs

### PR DESCRIPTION
**Pull request by stephan-strate**
Extension for #641.

### CODE REVIEW

**Review by Ladicek — APPROVED** [review]

**Comment by manofthepeace** [comment]
Funny coincidence, I did the almost same change last Friday locally. One thing I saw is that in the case where it is not the CompletionStage, the counter should be incremented by 1 in the log, since it starts at 0, so in the log it wont ever reach the maxAttempts.

For completionStage I had this;
`LOG.debugf("%s invocation failed, retrying (attempt %d of %d)", description, counter, maxRetries);`

for the non completionStage I had this;
`LOG.debugf("%s invocation failed, retrying (attempt %d of %d)", description, counter+1, maxRetries);`

otherswise if you have let say 3 retries , 2 of 3 will be the last which I think is counterintuitive. For the completion stage, it seems ok since it enters the if statement above the log when the counter is at 1, when it is actually doing the first retry.

So I think, in the current pr,  one counter will start at 1, and reach the maxRetries, the other will start at 0 and reach MaxRetries-1

what do you guys think? I can Pr it if it makes sense

**Comment by Ladicek** [comment]
I admit I didn't try it, but I'm looking at the code and `counter == 0` in `Retry`, as well as `attempt == 0` in `CompletionStageRetry` is the initial invocation, so they seem consistent to me and no increment should be necessary...

**Comment by manofthepeace** [comment]
I just ran it and it works as it should. Sorry for the noise.

File: implementation/core/src/main/java/io/smallrye/faulttolerance/core/timeout/CompletionStageTimeout.java [hunk 1]
```diff
@@ -34,7 +34,7 @@ private CompletionStage<V> doApply(InvocationContext<CompletionStage<V>> ctx) {
         AtomicBoolean completedWithTimeout = new AtomicBoolean(false);
         Runnable onTimeout = () -> {
             if (completedWithTimeout.compareAndSet(false, true)) {
-                LOG.debug(description + " invocation timed out");
+                LOG.debug(description + " invocation timed out (" + timeoutInMillis + "ms)");
```
Comment (rhusar) [inline implementation/core/src/main/java/io/smallrye/faulttolerance/core/timeout/CompletionStageTimeout.java:37]: BTW missing space before 'ms' - fixed in https://github.com/smallrye/smallrye-fault-tolerance/pull/685

**Comment by rhusar** [comment]
This change adds a lot of unnecessary String concatenations – oped PR to fix https://github.com/smallrye/smallrye-fault-tolerance/pull/685

**Pull request merged**